docs/spec_draft.md

```diff
@@ -1406,6 +1407,61 @@ and produces a `result` tensor. More formally,
 ```
 
 [Back to Ops](#index-of-ops)
+
+## stablehlo.rng
+
+### Semantics
+
+Constructs a `result` tensor of a given shape `shape` with random numbers
+generated using the `rng_distribution` algorithm. If `rng_distribution` $=$
+`UNIFORM`, then the random numbers are generated following the uniform
+distribution over the interval [`a`, `b`). If `rng_distribution` $=$ `NORMAL`,
+then the random numbers are generated following the normal distribution with
+mean = `a` and standard deviation = `b`.
+
+If `rng_distribution = UNIFORM` and `b` $\le$ `a`, the result is
+implementation-defined.
+
+### Inputs
+
+| Name               | Type                                                 |
+|--------------------|------------------------------------------------------|
+| `a`                | constant of integer, boolean, or floating-point type |
+| `b`                | constant of integer, boolean, or floating-point type |
+| `shape`            | 1-dimensional tensor of integer type                 |
```

Comment:
Let's call this "1-dimensional tensor constant of type `si64`" for now, because otherwise it is unclear what C2 means, and also update the `shape` in examples to say `i64` instead of `i32`. I'll sort this out later in the dynamism RFC.